internal/pkg/cli/secret_init.go

```diff
@@ -117,6 +149,80 @@ func (o *secretInitOpts) Ask() error {
 
 // Execute creates the secrets.
 func (o *secretInitOpts) Execute() error {
+	secretName := o.name
+	secretValues := o.values
+
+	if o.inputFilePath != "" {
+		var err error
+		secretName, secretValues, err = o.parseFile()
+		if err != nil {
+			return err
+		}
+	}
+
+	return o.putSecrets(secretName, secretValues)
+}
+
+func (o *secretInitOpts) putSecrets(secretName string, values map[string]string) error {
+	for envName, value := range values {
+		err := o.putSecret(secretName, envName, value)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (o *secretInitOpts) putSecret(secretName, envName, value string) error {
+	client, err := o.configureSecretPutter(envName)
+	if err != nil {
+		return err
+	}
+
+	tags := make(map[string]string)
+	for k, v := range o.resourceTags {
+		tags[k] = v
+	}
+	tags[deploy.AppTagKey] = o.appName
+	tags[deploy.EnvTagKey] = envName
+
+	name := fmt.Sprintf(fmtSecretParameterName, o.appName, envName, secretName)
+	in := ssm.PutSecretInput{
+		Name:      name,
+		Value:     value,
+		Overwrite: o.overwrite,
+		Tags:      tags,
+	}
+
+	out, err := client.PutSecret(in)
+	if err != nil {
+		var targetErr *ssm.ErrParameterAlreadyExists
+		if errors.As(err, &targetErr) {
+			log.Infoln(fmt.Sprintf("Secret %s already exists. Do not overwrite.\n", name))
+			return nil
+		}
+		return fmt.Errorf("put secret %s in environment %s: %w", secretName, envName, err)
+	}
+
+	version := aws.Int64Value(out.Version)
+	if version != 1 {
```

Comment:
```suggestion
	if o.overwrite {
```

Reply:
The concern is that even if `o.overwrite` is true, it does not necessarily mean that the value is overwritten. 

A user can say `secret init --name db-password --values test=test,prod=prod --overwrite", while "/copilot/app/test/secrets/db-password" exists and ""/copilot/app/prod/secrets/db-password" does not.

In this case only "/copilot/app/test/secrets/db-password" is overwritten.